### ISSUE TITLE
Training code and pre-trained model?

### ISSUE THREAD

**Issue by sxjzwq** [Prof-Lu-Cewu/Visual-Relationship-Detection#1]
Hi, 

Thanks for the code. It is a great work!

But can you also provide the training code? Then we can train our own W and b in Eq. (2).
And is that possible to provide your pre-trained model for the object detection and predicates prediction model, so we can extract the 'objectDetRCNN.mat' and 'UnionCNNfeaPredicate.mat' for our own images.

Thanks!

**Comment by erobic** [comment]
@sxjzwq Were you able to train CNN to classify predicates? Paper states: "Similarly, we train a second CNN (VGG net [44]) to classify each of our K = 70 predicates using the union of the bounding boxes of the two participating objects in that relationship". I am just wondering how much accuracy we can receive for this CNN.

**Comment by Prof-Lu-Cewu** [comment]
Yes, we are about to train CNN to classify predicates,

but the accuracy is not good enough.

---

发件人: Robik Shrestha notifications@github.com
发送时间: 2016年10月8日 13:37
收件人: Prof-Lu-Cewu/Visual-Relationship-Detection
主题: Re: [Prof-Lu-Cewu/Visual-Relationship-Detection] Training code and pre-trained model? (#1)

@sxjzwqhttps://github.com/sxjzwq Were you able to train CNN to classify predicates? Paper states: "Similarly, we train a second CNN (VGG net [44]) to classify each of our K = 70 predicates using the union of the bounding boxes of the two participating objects in that relationship". Not sure how accurate this CNN would be.

―

**Comment by sxjzwq** [comment]
@erobic We didn't try to train a CNN to classify predicates. However, we tried to train a CNN to predict the whole relationships directly. We pre-build a vocabulary with 15,000 elements. In this vocabulary, the label is a triple such as <bag, on, table>, which means each label is a relationship instance. So <bag, on, table> and <bag,under,table> are two labels. We treat this as a multi-label classification problem and train a VGG net on the Visual Genome dataset. This sounds crazy but we actually got some reasonable results and we use it as our baseline. The drawbacks of this way is that 15000 relationships only cover around 70% relationships in the Visual Genome.

I still wish @Prof-Lu-Cewu can publish their training code, then we can try it on VG or other datasets.

**Comment by erobic** [comment]
@Prof-Lu-Cewu  @sxjzwq 
Thank you for the reply!
I am actually working on a project to visualize predicates. Basically I want to generate images for predicates like "rides", "skates on" etc.
For this, I thought I could train a CNN with examples of predicates and then generate images, but the accuracies are not good enough, so I am wondering if I should really proceed forward to generating images.

**Comment by ronghanghu** [comment]
> I still wish @Prof-Lu-Cewu can publish their training code, then we can try it on VG or other datasets.

+1

**Comment by liqing-ustc** [comment]
Ask for the training code +1

**Comment by jesiws** [comment]
+1  
Many thanks!!!

**Comment by wtliao** [comment]
ask for complete training code, too. It's really a great work. I'm trying to build the whole framework, but to many problems occurred. So I wish the sharing of the code to help me understand the proposed approach better.

Many thanks!

**Comment by yanxp** [comment]
Ask for the training code +1

**Comment by nikita0511** [comment]
asking for training code +1

**Comment by singhanj13** [comment]
Asking for the training code +1 !

**Comment by jamesben6688** [comment]
Asking for the training code +1

**Comment by Narcissuscyn** [comment]
Asking for the training code +1

thanks very much!

**Comment by fomalhaut-b** [comment]
I know this thread is way too old but.......training code would be great! 

+1!

Thanks, awesome work

**Comment by alice-cool** [comment]
> I know this thread is way too old but.......training code would be great!
> 
> +1!
> 
> Thanks, awesome work

You can find new methods UVTransE **https://github.com/zhung2/uvtranse**

**Comment by samahwaleed** [comment]
Asking for the training code

**Comment by YOUNG-bit** [comment]
Asking for the training code!!
